From 39c1656520f811114d086426769ff01685c01048 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Thu, 17 Jun 2021 12:20:47 +0200
Subject: [PATCH 1028/1695] libpng: fix build issue when NEON is disabled

See Open issue https://github.com/glennrp/libpng/issues/289

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 91969147f5480e1a7401943d603beebf5ad24fdf libpng: fix build issue when NEON is disabled]
Change-Id: Ia876bb78cb2d0a226d955073368fdc9e8f80dad6
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391780
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Tested-by: Yannick FERTRE <yannick.fertre@foss.st.com>
---
 lib/libpng/pngpriv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libpng/pngpriv.h b/lib/libpng/pngpriv.h
index 583c26f9b..ed0034607 100644
--- a/lib/libpng/pngpriv.h
+++ b/lib/libpng/pngpriv.h
@@ -2117,7 +2117,7 @@ PNG_INTERNAL_FUNCTION(void, png_init_filter_functions_sse2,
 PNG_INTERNAL_FUNCTION(png_uint_32, png_check_keyword, (png_structrp png_ptr,
    png_const_charp key, png_bytep new_key), PNG_EMPTY);
 
-#if PNG_ARM_NEON_IMPLEMENTATION == 1
+#if (PNG_ARM_NEON_OPT > 0) && (PNG_ARM_NEON_IMPLEMENTATION == 1)
 PNG_INTERNAL_FUNCTION(void,
                       png_riffle_palette_neon,
                       (png_structrp),
-- 
2.47.2

